src/ast/passes/semantic_analyser.cpp

```diff
@@ -133,7 +133,28 @@ void SemanticAnalyser::visit(Identifier &identifier)
     identifier.type = CreateInt(
         std::get<0>(getIntcasts().at(identifier.ident)));
   }
+  else if (func_ == "nsecs")
+  {
+    identifier.type = CreateTimestampMode();
+    if (identifier.ident == "boot")
+    {
+      identifier.type.ts_mode = TimestampMode::boot;
+    }
+    else if (identifier.ident == "tai")
+    {
+      identifier.type.ts_mode = TimestampMode::tai;
+    }
+    else if (identifier.ident == "sw_tai")
+    {
+      identifier.type.ts_mode = TimestampMode::sw_tai;
+    }
+    else
+    {
+      goto err;
+    }
```

Comment:
I think that transforming this into a `switch` would make it more resistant to errors - the static analyser should report when an enum case is not covered.

Reply:
C++ should not be able to perform `switch case` on strings, maybe I failed to understand what you mean. Thanks.

Reply:
Ah, my bad, sorry.